manager/integration/tests/common.py

```diff
@@ -4598,3 +4602,51 @@ def restore_backup_and_get_data_checksum(client, core_api, backup, pod,
                                      'default')
 
     return data_checksum, output, restore_pod_name
+
+
+def generate_support_bundle(case_name):
+    """
+        Generate support bundle into folder ./support_bundle/case_name.zip
+
+        Won't generate support bundle if current support bundle count
+        greate than MAX_SUPPORT_BINDLE_NUMBER.
+        Args:
+            case_name: support bundle will named case_name.zip
+    """
+
+    os.makedirs("support_bundle", exist_ok=True)
+    file_cnt = len(os.listdir("support_bundle"))
+
+    if file_cnt >= MAX_SUPPORT_BINDLE_NUMBER:
+        warnings.warn("Ignoring the bundle download because of \
+                            avoiding overwhelming the disk usage.")
+        return
+
+    # Use API gen support bundle
+    client = get_longhorn_api_client()
+    url = client._url.replace('schemas', 'supportbundles')
+    data = {'description': case_name, 'issueURL': case_name}
+    res = requests.post(url, json=data).json()
+
+    id = res['id']
+    name = res['name']
+
+    support_bundle_url = '{}/{}/{}'.format(url, id, name)
+    for i in range(RETRY_EXEC_COUNTS):
+        res = requests.get(support_bundle_url).json()
+
+        if res['progressPercentage'] == 100:
+            break
+        else:
+            time.sleep(RETRY_INTERVAL_LONG)
+
+    # Download support bundle
+    download_url = '{}/download'.format(support_bundle_url)
```

Comment:
If timeout to wait for the support bundle ready, should have a log then ignore downloading.

Reply:
HI @innobead ,

I did test on local environment to simulate timeout scenario, current code `r.raise_for_status()` will catch timeout error as below log and ignore download. 
`HTTPConnectionPool(host='10.10.10.100', port=80): Max retries exceeded with url: /download (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f2941b91850>, 'Connection to 10.10.10.100 timed out. (connect timeout=10)'))`

If there have anything I missed, I will update asap, thank you.

Reply:
I mean, if running out to wait for `res['progressPercentage'] == 100`, you don't need to do the bundle download because it should not be there.

Reply:
Hi @innobead , 

I modified code to skip download when `res['progressPercentage'] != 100` for review, thank you